src/main/java/com/adamk33n3r/runelite/watchdog/EventHandler.java

```diff
@@ -189,26 +191,54 @@ private void onItemContainerChanged(ItemContainerChanged itemContainerChanged) {
-                        .findFirst()
-                        .ifPresent(groups -> this.fireAlert(inventoryAlert, groups));
-                }
-            });
+        Item[] items = itemContainerChanged.getItemContainer().getItems();
+        long itemCount = Arrays.stream(items).filter(item -> item.getId() > -1).count();
+        Map<Integer, Integer> allItems = new HashMap<>();
+        Arrays.stream(items)
+                .forEach(item -> allItems.merge(item.getId(), item.getQuantity(), Integer::sum));
+        // Skip firing alerts if there are no previous items, since we just logged in. Even an empty inventory will have a map of -1 itemIds.
+        if (!this.previousItemsTable.isEmpty()) {
+            this.alertManager.getAllEnabledAlertsOfType(InventoryAlert.class)
+                    .forEach(inventoryAlert -> {
+                        if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.FULL && itemCount == 28) {
+                            this.fireAlert(inventoryAlert, inventoryAlert.getInventoryAlertType().getName());
+                        } else if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.EMPTY && itemCount == 0) {
+                            this.fireAlert(inventoryAlert, inventoryAlert.getInventoryAlertType().getName());
+                        } else if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.ITEM) {
+                            allItems.entrySet().stream()
+                                    .filter(itemWithCount -> itemWithCount.getKey() != -1 && (inventoryAlert.getItemQuantity() == 0 || itemWithCount.getValue() == inventoryAlert.getItemQuantity()))
+                                    .map(itemWithCount -> this.matchPattern(inventoryAlert,
+                                            this.itemManager.getItemComposition(itemWithCount.getKey()).getName()))
+                                    .filter(Objects::nonNull)
+                                    .forEach(groups -> this.fireAlert(inventoryAlert, groups));
+                        } else if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.ITEM_CHANGE) {
+                            allItems.entrySet().stream()
+                                    .filter(itemWithCount -> {
+                                        if (itemWithCount.getKey() == -1) {
+                                            return false;
+                                        }
+                                        int change = itemWithCount.getValue() - this.previousItemsTable.getOrDefault(
+                                                itemWithCount.getKey(), 0);
+                                        // This would trigger if an item's quantity hasn't changed, but something else in the inventory triggered an inventory change event
+                                        if (inventoryAlert.getItemQuantity() == 0 && change == 0) {
```

Comment:
What is the use case for this one? What does it mean to set `itemQuantity` to `0`? Wouldn't this mean that it would fire every time your inventory changes but its _not_ the thing you typed?

Reply:
> Wouldn't this mean that it would fire every time your inventory changes but its not the thing you typed?

Yup exactly.

I'm not sure the use-case exactly, but I had to make a choice here if I wanted to support `0` as an option, or make it do nothing. I figured that it might as well be consistent in behavior even if it will never be used. That way it doesn't need to be explained why zero is not a valid value.

Reply:
If that's the case, I think that should be added to the tooltip. Otherwise, it would be confusing what it's doing.

Reply:
Really? That seems like the intuitive behavior for the zero case, it alerts when the delta of that item is zero (nothing changed).

I'll figure out how to document that.

Reply:
Would it be less confusing if zero just did nothing?

Reply:
Well the whole premise of the alert is "when this item amount is changed positive or negative" so that's why I see it as confusing that you can detect when "anything **but** this item is changed in either direction". You could also trigger that with a negative regex (e.g. **not** Coin).

It would certainly be less confusing if `0` did nothing, but I think it would still be weird that it exists. Too bad you can't like...disallow `0` from being selected. If the tooltip says `0 does nothing` and maybe it defaults to `1` that'd be sufficient.

Reply:
Actually, looking back at the code, I handle `0` in the `ITEM` type to mean "fire every time your inventory changes and you have this item in your inventory". Would you say that's accurate? So, maybe that's really the same case you got going here.

Reply:
Yeah, for the current Inventory Quantity alert, 0 triggers every time.

Reply:
Then I'd say this behavior is ok. But like I said it would be nice to describe it in a tooltip! While you're there, maybe in the current one too :P

Otherwise, LGTM!

Reply:
Done!